engine/src/windowing/window.cpp

```diff
@@ -2,29 +2,34 @@
 
 #include "windowing/window.hpp"
 
+#include "graphics/device.hpp"
+#include "logger/log.hpp"
+
 #include <GLFW/glfw3.h>
 
+
 #include <utility>
 
 namespace gen {
 
     Window::Window(int w, int h, std::string title) : m_width{w}, m_height{h}, m_title{std::move(title)} { // NOLINT(cppcoreguidelines-pro-type-member-init)
-        initWindow();
+        init();
```

Comment:
Why is this a separate function if it's only ever called once in the constructor?